server/neptune/adhoc/adhocexecutionhelpers/adhoc_execution_params.go

```diff
@@ -1,15 +1,32 @@
 package adhoc
 
 import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/neptune/adhoc/adhocgithubhelpers"
 	"github.com/runatlantis/atlantis/server/neptune/workflows/activities/github"
 	"github.com/runatlantis/atlantis/server/neptune/workflows/activities/terraform"
 )
 
 type AdhocTerraformWorkflowExecutionParams struct {
-	AtlantisRoot  string
-	AtlantisRepo  string
-	Revision      string
-	TerraformRoot terraform.Root
-	GithubRepo    github.Repo
+	Revision       string
+	TerraformRoots []terraform.Root
+	GithubRepo     github.Repo
 	// Note that deploymentID is used in NewWorkflowStore(), but we don't care about that in adhoc mode so can leave it blank
 }
+
+func (a *AdhocTerraformWorkflowExecutionParams) ConstructAdhocExecParamsWithRootCfgBuilderAndRepoRetriever(ctx context.Context, repoName string, revision string, githubRetriever *adhocgithubhelpers.AdhocGithubRetriever) error {
```

Comment:
Not sure I'm following this function, can you construct the AdhocTerraformWorkflowExecutionParams struct once you obtain all of the fields as opposed to mutating it later on?